src/main/java/net/javacrumbs/completionstage/SimpleCompletionStage.java

```diff
@@ -30,17 +29,29 @@
  * {@link java.util.concurrent.CompletionStage} implementation.
  * Keeps the result or callbacks in {@link net.javacrumbs.completionstage.CallbackRegistry}
  */
-class SimpleCompletionStage<T> extends CompletionStageAdapter<T> implements CompletableCompletionStage<T> {
+public class SimpleCompletionStage<T> extends CompletionStageAdapter<T> implements CompletableCompletionStage<T> {
```

Comment:
Protected would have been better, it tells that the class should not be used directly by can be extended. If it's public, we do not need the CompletionStageFactory